kernel/src/platform/mpu.rs

```diff
@@ -93,20 +93,30 @@ pub trait MPU {
     /// current state to help with debugging.
     type MpuConfig: Default + Display = MpuConfigDefault;
 
-    /// Enables the MPU.
+    /// Clears the MPU.
+    ///
+    /// This function will clear any access control enforced by the
+    /// MPU where possible. This should be called early in the Tock
+    /// init process to reset any configuration from previous stages.
+    /// On some hardware it is impossible to reset the MPU after it has
+    /// been locked, in this case this function wont change those regions.
+    fn clear_mpu(&self) {}
+
+    /// Enables the MPU for userspace apps.
     ///
     /// This function must enable the permission restrictions on the various
     /// regions protected by the MPU.
-    fn enable_mpu(&self) {}
+    fn enable_app_mpu(&self) {}
 
-    /// Disables the MPU.
+    /// Disables the MPU for userspace apps.
     ///
-    /// This function must completely disable any access control enforced by the
-    /// MPU. This will be called before the kernel starts to execute as on some
+    /// This function must disable any access control that was previously setup
+    /// for an app if it will interfere with the kernel.
```

Comment:
I don't understand what this comment means. It seems to contradict the second part (unmodified) of the comment. What does this do? What is "access control that was previously setup for an app if it will interfere with the kernel?" Does this mean "give access to kernel memory"?

Reply:
The idea is to "undo" any app specific changes ONLY if they interfere with the running of the kernel.

So if we support some new architecture that requires disabling the MPU before we can run the rest of the kernel then we should do that here.